com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/ConfigurationDoneRequestHandler.java

```diff
@@ -63,7 +68,20 @@ private void handleDebugEvent(DebugEvent debugEvent, IDebugSession debugSession,
         Event event = debugEvent.event;
         boolean isImportantEvent = true;
         if (event instanceof VMStartEvent) {
-            // do nothing.
+            if (context.isVmStopOnEntry()) {
+                EventRequestManager manager = debugSession.getVM().eventRequestManager();
+                ArrayList<MethodEntryRequest> legacy = new ArrayList<>(manager.methodEntryRequests());
+                manager.deleteEventRequests(legacy);
+                MethodEntryRequest request = manager.createMethodEntryRequest();
+                request.setSuspendPolicy(EventRequest.SUSPEND_EVENT_THREAD);
+                request.enable();
```

Comment:
Not sure about the performance of your implemenation. actually there is the second implementation. Because the main class is known, finding the `ReferenceType` from VM directly, then set a breakpoint at the location of main method.

Reply:
I  didn't know that. Seems your option is much better as VM could get the entry point directly! Let me have a look at it:)